app/controllers/krikri/providers_controller.rb

```diff
@@ -0,0 +1,19 @@
+module Krikri
+  # This models a provider that is stored in Marmotta.
+  class ProvidersController < ApplicationController
+    def index
+      @providers = Krikri::Provider.all
+    end
+
+    def show
+      if params[:set_session]
+        session[:current_provider] = params[:id]
+        redirect_to :back, provider: params[:id]
```

Comment:
This isn't behaving as expected.  I'm aiming to get it to redirect to the calling page, but with the added provider param.  Perhaps there's a better way of approaching this?